ert_shared/enkf_facade.py

```diff
@@ -35,3 +36,24 @@ def get_queue_config(self):
 
     def get_number_of_iterations(self):
         return self._enkf_main.analysisConfig().getAnalysisIterConfig().getNumIterations()
+
+    def get_observations(self):
+        return self._enkf_main.getObservations()
+
+    def get_impl_type_name_for_obs_key(self, key):
+        return self._enkf_main.getObservations()[key].getImplementationType().name
+
+    def get_current_fs(self):
+        return self._enkf_main.getEnkfFsManager().getCurrentFileSystem()
+
+    def get_impl(self):
```

Comment:
Is this not a massive leakage in the facade? Could we not move the parts that require the `EnKFMain` behind the facade instead?

Reply:
Yes it is, the problem is that several functions, for example the `SummaryCollector` requite `EnkFMain`, would be good to rewrite them to not require `EnkFMain` and have explicit input instead, but will require a substantial API change.

Reply:
How about if we moved the calls like:

```python
GenDataCollector.loadGenData(
             facade.get_impl(), case_name, data_key, time_step
         )
```

into the facade? Then, `EnKFMain` remains inside the facade. Yes, the interface of the facade grows, but the entire point of it is to be transparent about the usage of `EnKFMain`.

Reply:
Extending the facade with these loaders bloats the facade, imo. I agree that the leak must be fixed, but then we could move the data loaders out of libres entirely and into ERT's new data API (where MeasureData is slated to live). Then refactor the data loaders to use the facade, keeping the facade focused and light.

WDYT?

Reply:
> Extending the facade with these loaders bloats the facade, imo.

I completely agree. At the same time, is not the facade meant to be a true, yet as confined as possible, representation of the functionality we use in `libres`?

Personally I would prefer to not have the leak, bloat the facade and make the problem very visible. Then move the data loaders as you say, and then removing the similar functionality in the facade. Note that unless the leak would remain, we would have to break the API of the facade anyway in this process.

That would be my preference. But if the rest of you feel differently, I'm fine with doing it differently. What do you think @oysteoh, the originator of the facade?

Reply:
I agree to most of the things - the purpose was to separate and hide enkf main making it possible refactor that one without breaking ERT multiple places . I tried moving all existing enkf main specifics in ert to the facade and use that api - but it’s a mess... i would guess this has to be an interative process which initially needs to be implemented while holding your nose - but at each iteration will be better and better ... i can have a closer look tomorrow when i have more time - maybe i’m able to provide some more opinions

Reply:
Okay, I see what you mean. We need to rename the facade then, to `LibresFacade`, otherwise we're going to be confused somewhere down the road.

For the latter part (moving loaders), I've created https://github.com/equinor/ert/issues/581

Reply:
Yes , absolutely ! Naming is confusing and was about to change in my abandoned pr . Please feel free!